### ISSUE TITLE
🛠️ collect additional metrics in Azure App Insights 

### ISSUE THREAD

**Issue by brindasasi** [GCTC-NTGC/gc-digital-talent#9375]
## 🛠️ Tooling

We would like to know 
1. whether the user came from GC Jobs, Google, or anywhere else
2. what is the role of the user

## 🕵️ Details

Add any additional details that could assist with the changes/additions being made.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt tooling was properly updated and remains functional.

- [ ] Criteria 1
- [ ] Criteria 2

**Comment by brindasasi** [comment]
do we have any decision on this ? @GCTC-NTGC/developers

**Comment by brindasasi** [comment]
as per the above linked doc 

> It isn't necessary to use the user's actual sign-in name. It only has to be an ID that is unique to that user. It must not include spaces or any of the characters ,;=|.

**Comment by petertgiles** [comment]
@gobyrne Is this something we could do?  Should do?

**Comment by brindasasi** [comment]
![image (10)](https://github.com/user-attachments/assets/393597b4-3155-4b26-bb81-fd185367e48f)
![image (9)](https://github.com/user-attachments/assets/58c3d6b6-42c0-49f5-8c39-be167843e6f8)

**Comment by tristan-orourke** [comment]
https://talent-cloud.slack.com/archives/C0259G6KXJ6/p1723233611676579

**Comment by tristan-orourke** [comment]
I've looked at refUri, and I believe it tracks how a user moves from page to page within our app, but not where they came from before arriving at our app. Brinda has a draft PR which I'll try on Dev to see if it works differently.

**Comment by petertgiles** [comment]
I haven't spent too much time in App Insights but our app service HTTP logs do have Referrer available as well.

![Image](https://github.com/user-attachments/assets/3b5de3cb-5594-47d8-b845-65cf39dccb56)

**Comment by brindasasi** [comment]
Btw, @tristan-orourke  The first entry in @petertgiles's screenshot is actually from the  gc jobs site. Look at the url in below screenshot of gc jobs 
![Image](https://github.com/user-attachments/assets/b45eeb32-bff5-4b67-80e4-033f2e0288d7)

**Comment by petertgiles** [comment]
I was thinking about those `undefined`s. I think it makes sense that most hits don't have a referrer.  That would cover anyone typing in the URL, launching it from a bookmark, or launching it from link in an email, right?

**Comment by brindasasi** [comment]
yeah .. I combed through them through the http logs. Looks like genuine ones and we are good with that. Will close this ticket and my pr as not needed anymore.
